skimage/filters/rank/core_cy.pyx

```diff
@@ -271,3 +286,293 @@ cdef void _core(void kernel(dtype_t_out*, Py_ssize_t, Py_ssize_t*, double,
+
+                r += 1  # pass to the next row
+                if r >= rows:
+                    break
+
+                # ---> north to south
+                for j in range(num_se_s):
+                    rr = r + se_s_r[j]
+                    cc = c + se_s_c[j]
+                    pp = p + se_s_p[j]
+                    if is_in_mask_3D(rows, cols, planes, rr, cc, pp,
+                                     mask_data):
+                        histogram_increment(histo, &pop, image[pp, rr, cc])
+
+                for j in range(num_se_n):
+                    rr = r + se_n_r[j] - 1
+                    cc = c + se_n_c[j]
+                    pp = p + se_n_p[j]
+                    if is_in_mask_3D(rows, cols, planes, rr, cc, pp,
+                                     mask_data):
+                        histogram_decrement(histo, &pop, image[pp, rr, cc])
+
+                kernel(&out[p, r, c, 0], odepth, histo, pop,
+                       image[p, r, c], n_bins, mid_bin, p0, p1, s0, s1)
+
+                # ---> east to west
+                for c in range(cols - 2, -1, -1):
+                    for j in range(num_se_w):
+                        rr = r + se_w_r[j]
+                        cc = c + se_w_c[j]
+                        pp = p + se_w_p[j]
+                        if is_in_mask_3D(rows, cols, planes, rr, cc, pp,
+                                         mask_data):
+                            histogram_increment(histo, &pop, image[pp, rr, cc])
+
+                    for j in range(num_se_e):
+                        rr = r + se_e_r[j]
+                        cc = c + se_e_c[j] + 1
+                        pp = p + se_e_p[j]
+                        if is_in_mask_3D(rows, cols, planes, rr, cc, pp,
+                                         mask_data):
+                            histogram_decrement(histo, &pop, image[pp, rr, cc])
+
+                    kernel(&out[p, r, c, 0], odepth, histo, pop,
+                           image[p, r, c], n_bins, mid_bin, p0, p1, s0, s1)
+
+                r += 1  # pass to the next row
+                if r >= rows:
+                    break
+
+                # ---> north to south
+                for j in range(num_se_s):
+                    rr = r + se_s_r[j]
+                    cc = c + se_s_c[j]
+                    pp = p + se_s_p[j]
+                    if is_in_mask_3D(rows, cols, planes, rr, cc, pp,
+                                     mask_data):
+                        histogram_increment(histo, &pop, image[pp, rr, cc])
+
+                for j in range(num_se_n):
+                    rr = r + se_n_r[j] - 1
+                    cc = c + se_n_c[j]
+                    pp = p + se_n_p[j]
+                    if is_in_mask_3D(rows, cols, planes, rr, cc, pp,
+                                     mask_data):
+                        histogram_decrement(histo, &pop, image[pp, rr, cc])
+
+                kernel(&out[p, r, c, 0], odepth, histo, pop, image[p, r, c],
```

Comment:
what is the `0` for in `out[p, r, c, 0]`?

Reply:
from 2D, will look into it, before that @ryanlu41 ?